lib/compat.php

```diff
@@ -125,3 +125,17 @@ function gutenberg_get_post_from_context() {
 	}
 	return get_post();
 }
+
+/**
+ * Reorders the priority on the_content filters for blocks and autoembeds
+ *
+ * See: https://core.trac.wordpress.org/ticket/46457
+ */
+function gutenberg_reorder_content_filters() {
+	global $wp_embed;
+	remove_filter( 'the_content', array( $wp_embed, 'autoembed' ), 8 );
+	remove_filter( 'the_content', 'do_blocks', 9 );
+	add_filter( 'the_content', 'do_blocks', 8 );
+	add_filter( 'the_content', array( $wp_embed, 'autoembed' ), 9 );
```

Comment:
Apparently these has the potential of breaking autop things, so I'd like some good testing for this PR. Also  cc @aduth since you're better aware of the potential issues.

Reply:
Also, this has the potential to affect dynamic blocks, so we need to check these too.

Reply:
Agreed. I was thinking that the tests from #11050 had those things covered now, but it's always possible we're missing things.

Reply:
At least within the block filtering itself, the handling of `wpautop` is pretty self-contained, and generally tolerant to adjustments in its own priority.

https://github.com/WordPress/wordpress-develop/blob/7b5d78f82f90a4a1274cba8bd078296461a6e768/src/wp-includes/blocks.php#L540-L545

This specific change also wouldn't have an impact so far as: Blocks filtering occurred before `wpautop` before these changes, and still occurs before `wpautop` _after_ these changes. Specifically, `wpautop` happens at the default priority (`10`).

https://github.com/WordPress/wordpress-develop/blob/7b5d78f82f90a4a1274cba8bd078296461a6e768/src/wp-includes/default-filters.php#L175

Relevant to my [previous comment](https://github.com/WordPress/gutenberg/pull/21043#issuecomment-610933446):

>All of this processing should be expected to occur before the default priority (`10`) since a filter at the default priority would expect to receive markup in roughly its "ready" state, at least so far as what is provided by core.

At first I wondered if this contradicted my statement. But before `default-filters.php` runs fairly early, even though it uses the same priority, since the filter is added early, it's likely still run long before any plugins would. Aside from the bugs and resulting changes here, it may have even been appropriate that `do_blocks` [a few lines above](https://github.com/WordPress/wordpress-develop/blob/7b5d78f82f90a4a1274cba8bd078296461a6e768/src/wp-includes/default-filters.php#L172) be registered with the default priority as well, since the order of the filter adds would have largely emulated the same effect.